### PR TITLE
Finished user profile page

### PR DESCRIPTION
**Pull request by blafri**
Hey Felix,
Just finished the user settings page so that users can now update their profile settings, cancel their account etc. I created a custom controller to handle this with edit and update actions as I wanted to do stuff a little different to how devise does it by default, and I thought that a new controller for those actions would be the best way.

I also added a time_zones table and set up a field in the users table to track their time zone. Figured we would need it since we were going to be doing some time dependant tasks and the users could be from different time zones than the default time zone for the server.

Also I setup two other fields on the user table for email reminders of tasks. One is for the user to opt into receiving reminder emails and the other one is so the user can set how old the task must be to show up on the summary email of the users tasks. So for instance the user will be able to select 2 and he will get daily summary emails of of all outstanding tasks that are 2 days from expiring.

Let me what you think and if you can see anything that can be improved.

Thanks,
Blayne

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -4,63 +4,20 @@
...
-
-  # Example resource route within a namespace:
-  #   namespace :admin do
-  #     # Directs /admin/products/* to Admin::ProductsController
-  #     # (app/controllers/admin/products_controller.rb)
-  #     resources :products
-  #   end
+  # Devise routes we skip the registration routes and them recreate them manually
+  # leaving out edit and update as those are implemented in the users controller
+  devise_for :users, skip: :registrations
```
Comment (felixclack) [inline config/routes.rb]:
```ruby
devise_for :users, controllers: { registrations: :users_controller }
```

Then in your class

```ruby
class UsersController < Devise::RegistrationsController
  def edit
    ...
  end
end
```

**Comment by felixclack** [comment]
Looks good.

**Comment by blafri** [comment]
Ok Felix,
I changed it to inherit from Devise::RegistrationsController. Really saw the power of writing all those tests while re-factoring the code. Made it simple to find where the errors were and be confident that everything was working after.

Thanks,
Blayne

**Comment by felixclack** [comment]
:+1:

**Pull request merged**